redcalendar/events/templates/events/index.html

```diff
@@ -7,7 +7,7 @@
 	 <p>Where: {{ event.event_place}} </p>
 	 {{event.event_poster}}
 	 <p>When: {{ event.event_date }} </p>
-	 <img src="{{ event.event_poster}}" >
+	 <img src="{{ event.event_poster.url }}" >
```

Comment:
This was mainly the problem. You need to `event.event_poster` is File object. What we need is the `url` of that file object.